deps/ncrypto/ncrypto.cc

```diff
@@ -3080,6 +2943,69 @@ bool CipherCtxPointer::getAeadTag(size_t len, unsigned char* out) {
 
 // ============================================================================
 
+KdfCtxPointer KdfCtxPointer::FromName(std::string_view name) {
+  auto kdf = EVP_KDF_fetch(nullptr, name.data(), nullptr);
+  auto ctx = EVP_KDF_CTX_new(kdf);
+  return KdfCtxPointer{ctx};
```

Comment:
The KDF context stores a reference to the KDF, that's why we don't need to store it ourselves. It's also possible to fetch the KDF from the context if needed